### DISCUSSION TITLE
Filters in the app (for end user)

### DISCUSSION THREAD

**Discussion by andreapx** [Budibase/budibase#10103] [Help] [answered]
Hi all!
How can I add a filter in the app so that the users (in the _frontend_ ) can, for example, select a date and view only the row that are newer than the data picked?

**Comment by ConorWebb96** [comment, marked as answer]
Hey @andreapx,

If you are using a Budibase table this should be somewhat simple. All you would need to do is define filters matched against your date field within the data provider. (I've supplied some screenshots below to help explain this better.)

**Component breakdown**
<img width="262" alt="Screenshot 2023-03-27 at 08 29 10" src="https://user-images.githubusercontent.com/126772285/227871031-069ac9b5-babc-4b2c-b661-64db9e9ae72a.png">

**The defined filters within the data provider**
<img width="1071" alt="Screenshot 2023-03-27 at 08 21 47" src="https://user-images.githubusercontent.com/126772285/227870130-d46c4fb3-7e51-4f91-86e9-3322851ed9a4.png">

**The full binding for the second filter**
<img width="766" alt="Screenshot 2023-03-27 at 08 21 56" src="https://user-images.githubusercontent.com/126772285/227870140-d2aaf931-3d1b-43f5-83d0-6949ad537c9a.png">

**Gif of the finished product**
![Date filtering](https://user-images.githubusercontent.com/126772285/227871180-da4e2bd0-275b-4f15-a7bd-9129739c5db6.gif)

A good source for additional reading can be found here: https://docs.budibase.com/docs/adding-a-search-field#add-a-form-and-text-field

I hope this is helpful and answers your question.

**Reply by melohagan** [reply]
You can also do this a little easier with a Dynamic filter, but imo this approach has a nicer UX!
